test/unit/org/apache/cassandra/tools/JMXCompatabilityTest.java

```diff
@@ -0,0 +1,178 @@
+                                                    "org.apache.cassandra.metrics:type=Client,name=connectedThriftClients", // removed in CASSANDRA-11115
+                                                    "org.apache.cassandra.request:type=ReadRepairStage", // removed in CASSANDRA-13910
+                                                    "org.apache.cassandra.db:type=HintedHandoffManager", // removed in CASSANDRA-15939
+
+                                                    // Cas*Latency metrics missing
+                                                    ".*keyspace=system,scope=schema_aggregates,.*",
+                                                    ".*keyspace=system,scope=schema_columnfamilies,.*",
+                                                    ".*keyspace=system,scope=schema_columns,.*",
+                                                    ".*keyspace=system,scope=schema_functions,.*",
+                                                    ".*keyspace=system,scope=schema_keyspaces,.*",
+                                                    ".*keyspace=system,scope=schema_triggers,.*",
+                                                    ".*keyspace=system,scope=schema_usertypes,.*",
+
+                                                    // dropped tables
+                                                    ".*keyspace=system,(scope|table|columnfamily)=views_builds_in_progress.*",
+                                                    ".*keyspace=system,(scope|table|columnfamily)=range_xfers.*",
+                                                    ".*keyspace=system,(scope|table|columnfamily)=hints.*",
+                                                    ".*keyspace=system,(scope|table|columnfamily)=batchlog.*");
+        List<String> excludeAttributes = Arrays.asList("RPCServerRunning", // removed in CASSANDRA-11115
+                                                       "MaxNativeProtocolVersion");
+        List<String> excludeOperations = Arrays.asList("startRPCServer", "stopRPCServer", // removed in CASSANDRA-11115
+                                                       // nodetool apis that were changed,
+                                                       "decommission", // -> decommission(boolean)
+                                                       "forceRepairAsync", // -> repairAsync
+                                                       "forceRepairRangeAsync", // -> repairAsync
+                                                       "beginLocalSampling", // -> beginLocalSampling(p1: java.lang.String, p2: int, p3: int): void
+                                                       "finishLocalSampling" // -> finishLocalSampling(p1: java.lang.String, p2: int): java.util.List
+        );
+
+        diff(excludeObjects, excludeAttributes, excludeOperations, "test/data/jmxdump/cassandra-3.0-jmx.yaml");
+    }
+
+    @Test
+    public void diff311() throws Throwable
```

Comment:
Interesting about whether to handle renames - `HintedHandOffManager` --> `HintsService`

Reply:
mixed feeling about this.  Right now I did what was needed to get these tests to pass; I am not saying that these changes are ok.

As far as I could tell, we wanted to move from HintedHandOffManager to HintsService a long time ago, but wasn't clear if all metrics are present in `HintsService` as 4.0 renamed and didn't have alias support for this code like we do for tables... We can always check what was present in `HintsService` (all gold files would show what's there for each version) to make sure nothing got dropped.

now, if we wanted to diff just those two that wouldn't be hard as it would be to just extract those two objects and diff them.


tested this real quick

```
@Test
    public void diff311HintService() throws Throwable
    {
        setupStandardTables();

        try (InputStream left = new FileInputStream("test/data/jmxdump/cassandra-3.11-jmx.yaml");
             InputStream right = new FileInputStream(TMP.getRoot().getAbsolutePath() + "/out.yaml"))
        {
            Map<String, JMXTool.Info> three11 = JMXTool.Diff.Format.yaml.load(left);
            Map<String, JMXTool.Info> trunk = JMXTool.Diff.Format.yaml.load(right);

            JMXTool.Info hintedHandOffManager = three11.get("org.apache.cassandra.db:type=HintedHandoffManager");
            JMXTool.Info hintService = trunk.get("org.apache.cassandra.hints:type=HintsService");

            JMXTool.Diff differ = new JMXTool.Diff();
            differ.diff(ImmutableMap.of("a", hintedHandOffManager), ImmutableMap.of("a", hintService));
        }
    }
```

```
Difference found in attribute or operation
a	operation not in right:
	deleteHintsForEndpoint(p1: java.lang.String): void
	listEndpointsPendingHints(): java.util.List
	pauseHintsDelivery(p1: boolean): void
	scheduleHintDelivery(p1: java.lang.String): void
	truncateAllHints(): void
a	operation not in left:
	deleteAllHints(): void
	deleteAllHintsForEndpoint(p1: java.lang.String): void
	pauseDispatch(): void
	resumeDispatch(): void

```